### ISSUE TITLE
Error with state_summary() tidy = TRUE

### ISSUE THREAD

**Issue by slarge** [mps9506/rATTAINS#25]
Hi -- I'm interested in exploring EPA ATTAINS records for multiple states in the Northeast US. 
```

## R version 4.1.0
library(rATTAINS) ## 0.1.3.9999
library(dplyr) ## 1.0.7

domain_values(domain_name = "OrgStateCode") %>%
  filter(code %in% c("RI")) %>%
  pull(context)
## [1] "RIDEM" "EPA"  

state_summary(organization_id = "RIDEM") 
## Error: Can't coerce element 10 from a list to a character

## Whereas, when the code is not tidy it returns data
head(state_summary(organization_id = "RIDEM", tidy = FALSE))
```

Please let me know if you need additional information to make sure the code above is reproducible.

**Comment by mps9506** [comment]
Hi! Sorry about the bug. This confirms that the the ATTAINS API doesn't always return the same values for all states, which is something I was afraid off. I'll need to come up with some way to identify the values before parsing the JSON data in each function. It might take me some time to come up with a solution to incorporate in the package.

That said, here is a work-around (sorry that is is so verbose):

```r
library(rATTAINS)
library(dplyr)
library(purrr)
library(tidyjson)
library(janitor)

domain_values(domain_name = "OrgStateCode") %>%
  filter(code %in% c("RI")) %>%
  pull(context)


state_summary(organization_id = "TCEQMAIN")


data <- state_summary(organization_id = "RIDEM", tidy = FALSE)

data |> 
  enter_object("data") |> 
  spread_values(organizationIdentifier = jstring("organizationIdentifier"),
                organizationName = jstring("organizationName"),
                organizationTypeText = jstring("organizationTypeText")) |> 
  select(-c(.data$document.id)) |> 
  enter_object("reportingCycles") |> 
  gather_array() |> 
  spread_values(reportingCycle = jstring("reportingCycle")) |> 
  select(-c(.data$array.index)) |> 
  enter_object("waterTypes") |> 
  gather_array()  |> 
  spread_values(waterTypeCode = jstring("waterTypeCode"),
                unitsCode = jstring("unitsCode")) |> 
  select(-c(.data$array.index)) |> 
  enter_object("useAttainments") |> 
  gather_array() |> 
  spread_values(useName = jstring("useName"),
                fullySupporting = jstring("Fully Supporting"),
                fullySupportingCount = jstring("Fully Supporting-count"),
                notAssessed = jstring("Not Assessed"),
                notAssessedCount = jstring("Not Assessed-count")) |> 
  select(-c(.data$array.index)) |> 
  mutate(parameters = map(.data$..JSON, ~{
    .x[["parameters"]] %>% {
      tibble(parameterGroup = map(., "parameterGroup", .default = NA) %>% flatten_chr(),
             cause = map(., "Cause", .default = NA) %>% flatten_chr(),
             causeCount = map(., "Cause-count", .default = NA) %>% flatten_chr(),
             meetingCriteria = map(., "Meeting Criteria", .default = NA) %>% flatten_dbl(),
             meetingCriteriaCount = map(., "Meeting Criteria-count", .default = NA) %>% flatten_dbl(),
             insufficentInformation = map(., "Insufficient Information", .default = NA) %>% flatten_dbl(),
             insufficientInformationCount = map(., "Insufficient Information-count", .default = NA) %>% flatten_dbl()) %>%
        clean_names()
    }
  })) |> 
  as_tibble() |> 
  clean_names()
```

This should return:
```
# A tibble: 201 x 12
   organization_identifier organization_name organization_type_text reporting_cycle water_type_code units_code use_name   fully_supporting
   <chr>                   <chr>             <chr>                  <chr>           <chr>           <chr>      <chr>      <chr>           
 1 RIDEM                   Rhode Island      State                  2008            RIVER           Miles      Public Dr~ 6.082200        
 2 RIDEM                   Rhode Island      State                  2008            RIVER           Miles      Primary C~ 365.823700      
 3 RIDEM                   Rhode Island      State                  2008            RIVER           Miles      Fish and ~ 451.521500      
 4 RIDEM                   Rhode Island      State                  2008            RIVER           Miles      Fish Cons~ 9.946800        
 5 RIDEM                   Rhode Island      State                  2008            RIVER           Miles      Secondary~ 365.823700      
 6 RIDEM                   Rhode Island      State                  2008            RIVER           Miles      Shellfish~ NA              
 7 RIDEM                   Rhode Island      State                  2008            COASTAL         Miles      Secondary~ 78.620000       
 8 RIDEM                   Rhode Island      State                  2008            COASTAL         Miles      Fish Cons~ 78.620000       
 9 RIDEM                   Rhode Island      State                  2008            COASTAL         Miles      Fish and ~ NA              
10 RIDEM                   Rhode Island      State                  2008            COASTAL         Miles      Primary C~ 78.620000       
# ... with 191 more rows, and 4 more variables: fully_supporting_count <chr>, not_assessed <chr>, not_assessed_count <chr>,
#   parameters <list>
```

Leave this issue open and I will eventually get to fixing the issue properly.

**Comment by mps9506** [comment]
@mps9506 
https://github.com/mps9506/rATTAINS/blob/50420746dbecc021e8e3921261a5591649cb9946/R/state_summary.R#L117 sometimes causes an error when combined cycles are not reported (the API inconsistently provides no values for combined cycles or does not return combined cycles).
Likely options are to include some logic to test what values the API returns or use `spread_all()`. The downside of `spread_all()` is that we will get inconsistent data frame structure (both columns and types).

**Comment by mps9506** [comment]
Related to #31

**Comment by mps9506** [comment]
@slarge sorry for such a long wait on a fix. I have a development version in the dev-flatten branch that fixes this issue. I'm waiting on the tibblify package to mature before I release this version of rATTAINS to CRAN. You can install with the remotes package:
```r
remotes::install_github("mps9506/rATTAINS", ref = "dev-flatten")
```

```r
library(rATTAINS)
state_summary(organization_id = "RIDEM") 
```

```
# A tibble: 576 × 18
   organizati…¹ organ…² organ…³ repor…⁴ water…⁵ units…⁶ use_n…⁷ fully…⁸ fully…⁹ not_a…˟ not_a…˟ not_s…˟ not_s…˟ param…˟  cause
   <chr>        <chr>   <chr>   <chr>   <chr>   <chr>   <chr>     <dbl>   <int>   <dbl>   <int>   <dbl>   <int> <chr>    <dbl>
 1 RIDEM        Rhode … State   2008    RIVER   Miles   Public…    6.08       1    198.      69     NA       NA NA       NA   
 2 RIDEM        Rhode … State   2008    RIVER   Miles   Primar…  366.        79    746.     358    240.      69 PATHOG… 240.  
 3 RIDEM        Rhode … State   2008    RIVER   Miles   Fish a…  452.        92    702.     364    198.      50 NUISAN…   1.59
 4 RIDEM        Rhode … State   2008    RIVER   Miles   Fish a…  452.        92    702.     364    198.      50 ALGAL …  NA   
 5 RIDEM        Rhode … State   2008    RIVER   Miles   Fish a…  452.        92    702.     364    198.      50 NUTRIE…  28.5 
 6 RIDEM        Rhode … State   2008    RIVER   Miles   Fish a…  452.        92    702.     364    198.      50 CAUSE … 112.  
 7 RIDEM        Rhode … State   2008    RIVER   Miles   Fish a…  452.        92    702.     364    198.      50 ORGANI…  33.0 
 8 RIDEM        Rhode … State   2008    RIVER   Miles   Fish a…  452.        92    702.     364    198.      50 NUISAN…  40.1 
 9 RIDEM        Rhode … State   2008    RIVER   Miles   Fish a…  452.        92    702.     364    198.      50 DIOXINS   8.42
10 RIDEM        Rhode … State   2008    RIVER   Miles   Fish a…  452.        92    702.     364    198.      50 TOTAL …  10.0 
# … with 566 more rows, 3 more variables: cause_count <int>, insufficient_information <dbl>,
#   insufficient_information_count <int>, and abbreviated variable names ¹​organization_identifier, ²​organization_name,
#   ³​organization_type_text, ⁴​reporting_cycle, ⁵​water_type_code, ⁶​units_code, ⁷​use_name, ⁸​fully_supporting,
#   ⁹​fully_supporting_count, ˟​not_assessed, ˟​not_assessed_count, ˟​not_supporting, ˟​not_supporting_count, ˟​parameter_group
# ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names
```

**Comment by mps9506** [comment]
Should be resolved with the v1.0.0 update just released to CRAN. Closing this, if there are any issues feel free to reopen.
